lib/cext/preprocess.rb

```diff
@@ -352,7 +412,11 @@ def preprocess(line)
 
 def patch(file, contents, directory)
   if patched_file = PATCHED_FILES[File.basename(file)]
-    regexp = /^#{Regexp.escape(patched_file[:gem])}\b/
+    regexp = if patched_file[:ext_dir]
+               /^#{Regexp.escape(patched_file[:gem])}|#{Regexp.escape(patched_file[:ext_dir])}\b/
+             else
+               /^#{Regexp.escape(patched_file[:gem])}\b/
+             end
```

Comment:
What's the `file` here that did not match?
`parser/parser.c`? It sounds dangerous to match that.
Maybe we can File.expand_path to find the gem name in it.

Reply:
That and `generator.c`. The json gem internally organizes itself as three different extensions. I could make a pattern that includes both the gem name and the filename, but since I only have one example of a gem laid out like this I was trying to avoid overfitting.

Reply:
It seems much safer to me to make sure the gem name is in the expanded path of the C file.